### PR TITLE
Add snap packaging

### PR DESCRIPTION
**Pull request by om26er**
This PR adds snap packaging. A ‘snap’ is a universal Linux package, more reading here https://www.ubuntu.com/desktop/snappy

Once this PR is landed, we can setup automatic builds using https://build.snapcraft.io, once setup, for every new commit to master of this project, a new snap build kicks in and is released to the "edge" (or nightly) channel.

Note: I commit to the maintenance of axel as a snap package, so should there an issue arrive, I'll fix that.

### CODE REVIEW

**Comment by om26er** [comment]
The CI failure is probably not related to this change as its pretty isolated.

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -1,27 +1,43 @@
 name: axel
-version: '2.16.1'
-version-script: cat VERSION | cut -d ' ' -f1
+version: 'latest'
+version-script: cat $SNAPCRAFT_STAGE/VERSION | cut -d ' ' -f1
```
Comment (ismaell) [inline snap/snapcraft.yaml]: This could be `cut -d' ' -f1 $SNAPCRAFT_STAGE/VERSION`

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -1,27 +1,43 @@
...
 summary: Light command line download accelerator
 description: |
   Light command line download accelerator
-
-grade: stable
+grade: devel
```
Comment (ismaell) [inline snap/snapcraft.yaml]: Shouldn't this depend on the branch?

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -1,27 +1,43 @@
...
 confinement: strict
 
 parts:
   axel:
-    plugin: autotools
-    source: .
+    plugin: nil
+    override-build: |
+      snapcraftctl build
+      URL=$(curl https://api.github.com/repos/axel-download-accelerator/axel/releases/latest -s | jq .zipball_url -r)
```
Comment (ismaell) [inline snap/snapcraft.yaml]: Don't, simply don't do this sort of stuff.

Comment (ismaell) [reply snap/snapcraft.yaml]: There should be a better way to do nightly builds.

**Comment by ismaell** [comment]
Ping.

**Pull request closed without merging**